### PR TITLE
define source-git

### PR DESCRIPTION
**Pull request by TomasTomecek**
The first set of changes for source-git section of our docs.

This one provides a practical intro to what source-git is.

### CODE REVIEW

File: content/docs/source-git/_index.md [hunk 1]
```diff
@@ -4,6 +4,73 @@ draft: false
...
 
+Let's talk about what source-git actually is.
+
+We initiated source-git as an alternative to working in
+[dist-git](https://github.com/release-engineering/dist-git). The need was
+especially amplified after we realized that many projects already do this. The
+problem was that everyone created a workflow and tooling for themselves which
+made it really hard to share knowledge and onboard new projects. Our intention
+is to create an implementation which is highly configurable, easy to start with
+and provdes both command-line tooling and a service which is tightly
```
Comment (jpopelka) [inline content/docs/source-git/_index.md:15]:
```suggestion
and provides both command-line tooling and a service which is tightly
```

File: content/docs/source-git/_index.md [hunk 2]
```diff
@@ -4,6 +4,73 @@ draft: false
...
+integration with GitHub and GitLab.
```
Comment (jpopelka) [inline content/docs/source-git/_index.md:16]:
```suggestion
integrated with GitHub and GitLab.
```

File: content/docs/source-git/_index.md [hunk 3]
```diff
@@ -4,6 +4,73 @@ draft: false
...
+
+Source-git is a git repository which contains upstream sources and packaging
+files for a downstream distribtions, in our case Fedora Linux, CentOS Stream
```
Comment (jpopelka) [inline content/docs/source-git/_index.md:19]:
```suggestion
files for downstream distributions, in our case Fedora Linux, CentOS Stream
```

Comment (jpopelka) [inline content/docs/source-git/_index.md:19]: What do you mean by 'packaging files'? RPM spec file? A source-git repo usually contains also downstream patches.

File: content/docs/source-git/_index.md [hunk 4]
```diff
@@ -4,6 +4,73 @@ draft: false
...
+
+The basic requirement on a source-git repo is that one needs to be able to
+create a SRPM out of a current checkout. We need several steps to achieve such
+a thing:
+
+* RPM Spec file
+
+* Archive with sources
+
+* Spec file patches (if applicable)
```
Comment (jpopelka) [inline content/docs/source-git/_index.md:30]: Under "Spec file patch" I'd imagine a patch that modifies a spec file :-) I'd call it
- patches
- downstream patches
- patches to be applied in the spec file

(you can ignore this note if you don't like any of those or don't have any better)

File: content/docs/source-git/_index.md [hunk 5]
```diff
@@ -4,6 +4,73 @@ draft: false
...
+is to create an implementation which is highly configurable, easy to start with
+and provdes both command-line tooling and a service which is tightly
+integration with GitHub and GitLab.
+
+Source-git is a git repository which contains upstream sources and packaging
+files for a downstream distribtions, in our case Fedora Linux, CentOS Stream
+and Red Hat Enterprise Linux.
+
+The basic requirement on a source-git repo is that one needs to be able to
+create a SRPM out of a current checkout. We need several steps to achieve such
```
Comment (jpopelka) [inline content/docs/source-git/_index.md:23]:
```suggestion
create a SRPM out of a current checkout. We need several files to achieve such
```

File: content/docs/source-git/_index.md [hunk 6]
```diff
@@ -4,6 +4,73 @@ draft: false
...
+
+### Spec file
+
+I'm going to answer the elephant in the room right away: "Do we really need to
+have a spec file in our upstream repo, or a source-git repo?!" — No, you don't.
+We just need to have a way to obtain the spec file, so:
+* You can download it from anywhere you want
+* You can generate it
+* A template spec file can be filled in during a build process
+* Storing the spec file directly in the repository is obviously the most convenient
```
Comment (jpopelka) [inline content/docs/source-git/_index.md:41]:
```suggestion
have a spec file in our upstream repo, or a source-git repo?!" — No, you don't, even storing the spec file directly in the repository is the most convenient.
We just need to have a way to obtain the spec file, so, you can:
* download it from anywhere you want
* generate it
* fill in a template spec file during a build process
```

File: content/docs/source-git/_index.md (hunk 3 above)
Comment (TomasTomecek) [reply content/docs/source-git/_index.md:19]: no matter how many times I read it, I make so many stupid mistákos :D

File: content/docs/source-git/_index.md (hunk 5 above)
Comment (TomasTomecek) [reply content/docs/source-git/_index.md:23]: not sure about this one though it's true your suggestion is more concrete, mine is abstract

File: content/docs/source-git/_index.md (hunk 4 above)
Comment (TomasTomecek) [reply content/docs/source-git/_index.md:30]: I like a combination of 2) and 3) :D

File: content/docs/source-git/_index.md (hunk 3 above)
Comment (TomasTomecek) [reply content/docs/source-git/_index.md]: yes, precisely - okay, let me rephrase

**Comment by TomasTomecek** [comment]
@jpopelka thank you for such a thorough review, can we do round 2?

File: content/docs/source-git/_index.md [hunk 7]
```diff
@@ -4,6 +4,73 @@ draft: false
...
+
+Packit needs to be able to get an archive for the current git checkout. By
+default this is done using `git-archive` command and you can override this in
+any way you want:
+* Use autotools or any build system of your choice
+* Use the native way of how archives are being generated in a particular ecosystem
+* Script it yourself
+
+
+### Spec file patches
```
Comment (jpopelka) [inline content/docs/source-git/_index.md:55]: Do we want to change this to "Downstream patches to be applied in the spec file" as well?

**Review by jpopelka — APPROVED** [review]
